src/OpenTelemetry.Instrumentation.StartupHook/StartupHook.cs

```diff
@@ -65,4 +78,46 @@ private static string GetLoaderAssemblyLocation()
             throw;
         }
     }
+
+    private static string GetApplicationName()
+    {
+        var applicationName = string.Empty;
+        try
+        {
+            applicationName = AppDomain.CurrentDomain.FriendlyName;
+        }
+        catch (Exception ex)
+        {
+            StartupHookEventSource.Log.Error($"Error getting AppDomain.CurrentDomain.FriendlyName: {ex}");
+        }
+
+        return applicationName;
+    }
+
+    private static bool IsApplicationInExcludeList(string applicationName)
+    {
+        return GetExcludedApplicationNames().Contains(applicationName);
+    }
+
+    private static IList<string> GetExcludedApplicationNames()
+    {
+        var excludedProcesses = new List<string>();
+
+        var environmentValue = Environment.GetEnvironmentVariable("OTEL_PROFILER_EXCLUDE_PROCESSES");
```

Comment:
how about creating a static class with constants for all env vars? it can act as a backup for documentation. I like having "public contract" separated somewhere. We could also reuse this class e.g. in `ClearProfilerEnvironmentVariables`.

Reply:
I think that it might get tricky to use that approach for all environment variables, because we need to be careful about which dependencies the StartupHook takes on. So I don't think that we could share that code with the StartHook and the Tracer, but we could definitely share some code between the StartupHook and the tests. This would leave us with 3 locations where environment variables are defined: 1. the native profiler, 2. the managed tracer, 3. the startup hook. And some of the environment variables will be shared by more than one of those locations, such as the process inclusion/exclusion environment variables used by the native profiler and the startup hook.

Reply:
`Environment.GetEnvironmentVariable` can throw a `SecurityException` (probably only .NET Framework if permissions re locked down?) so I recommend putting the method call inside a try/catch block